src/ray.rs

```diff
@@ -49,6 +51,13 @@ where
     }
 }
 
+impl<S, P:Debug, V:Debug> Debug for Ray<S,P,V> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Ray")?;
+        <(&P,&V) as fmt::Debug>::fmt(&(&self.origin, &self.direction), f)
```

Comment:
I think this should be written differently, something like `f.debug_tuple("Ray).field(&self.origin).field(&self.direction)`

Reply:
I copied the Debug.fmt code from cgmath Vector3.
```
impl<S: fmt::Debug> fmt::Debug for Vector3<S> {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        try!(write!(f, "Vector3 "));
        <[S; 3] as fmt::Debug>::fmt(self.as_ref(), f)
    }
}
```

But your suggestion looks cleaner, doing the same for Line.

Reply:
yeah, a lot of the old code is suboptimal